src/config.c

```diff
@@ -140,6 +140,22 @@ int load_config(struct vpn_config *cfg, const char *filename)
 			strncpy(cfg->username, val, FIELD_SIZE - 1);
 		} else if (strcmp(key, "password") == 0) {
 			strncpy(cfg->password, val, FIELD_SIZE - 1);
+		} else if (strcmp(key, "set-dns") == 0) {
+			int set_dns = strtol(val, NULL, 0);
+			if (set_dns < 0 || set_dns > 1) {
+				log_warn("Bad set-dns in config file: \"%d\".\n",
+				         set_dns);
```

Comment:
I'd rather not display the `"%d"` value here. For instance if an user sets: `set-dns = true`, he/she will be returned: `Bad set-dns in config file: "0".` (because that's what `strtol` returns).